src/components/MapView/Legends/index.tsx

```diff
@@ -117,12 +146,53 @@ function LegendItem({
   opacity: initialOpacity,
   children,
   legendUrl,
+  exposure,
+  extent,
 }: LegendItemProps) {
+  const dispatch = useDispatch();
   const map = useSelector(mapSelector);
+  const analysisExposureLoading = useSelector(
+    isExposureAnalysisLoadingSelector,
+  );
+
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
   const [opacity, setOpacityValue] = useState<number | number[]>(
     initialOpacity || 0,
   );
 
+  const analysisResult = useSelector(analysisResultSelector);
+
+  const runExposureAnalysis = async () => {
+    if (!id) {
+      return;
+    }
+
+    if (!extent) {
+      return;
+    } // hasn't been calculated yet
```

Comment:
Can't we combine these?

Reply:
Included in a0b5b06310994f9da941bdfb6e90cddba68b58e6